File: Data Structures & Algorithms/anagram-groups/submission-1.cpp
#include <vector>
#include <unordered_map>
#include <string>
#include <algorithm>
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        std::unordered_map<std::string,std::vector<std::string>> dct;
        for (auto word: strs){
            auto sorted_word = word;
            std::sort(sorted_word.begin(),sorted_word.end());
            dct[sorted_word].push_back(word);
        }
        std::vector<std::vector<std::string>> res;
        for (auto &lst: dct){
            res.push_back(lst.second);
        }
        return res;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        dct = {}
        for num in nums:
            if num not in dct:
                dct[num] = 1
            else:
                dct[num]+=1
        lst = []
        for num in dct:
            lst.append([dct[num],num])
        lst.sort(reverse=True)
        final_lst = []

        for i in range(k):
            final_lst.append(lst[i][1])
        return final_lst
        

        